### ISSUE TITLE
Project changes are not persisted

### ISSUE THREAD

**Issue by arcady-zherdev** [bryntum/support#5355]
Crud Manager `load`-response includes `project` section but we don't build such section when persisting data to the server.

UPD: this should also work for the following snippet
```
project.applyProjectChanges(targetProject.changes)
```

**Comment by arcady-zherdev** [comment]
Initial progress pushed to `5355-save-project-changes` branch

**Comment by bmblb** [comment]
#219 appears related

**Comment by BrandonDH** [comment]
Hi Team, I'm following https://github.com/bryntum/support/issues/219 which I agree appears related - is there any development here? thanks!

**Comment by arcady-zherdev** [comment]
I started on the issue but had to switch to other more critical tasks.

The current workaround I can suggest to unblock you is something like this:

```js
class MyProjectModel extends ProjectModel {

    get changes() {
        let changes = super.changes;

        const projectChanges = this.modificationDataToWrite;

        // include project changes
        if (projectChanges) {
            changes = changes || {};
            changes.project = projectChanges;
        }

        return changes;
    }

}

new Gantt({
    project : new MyProjectModel({
        ...
    }),
    ...
})
```
You basically override default `ProjectModel` class `changes` getter to include project model changes.

After these changes the client will start sending project change to the server.

**Comment by BrandonDH** [comment]
Thanks for the update and temporary workaround @arcady-zherdev - is there a timeline for when this change will be in a release?

**Comment by dpk-suffolk** [comment]
@arcady-zherdev Thanks for the snippet but upon implementing this, we found that it does not match the expected sync response in the format: 
```
project : { updated/added/removed: [{startDate: Date}]}
```
Instead, we get back the entire project object in the sync response. Is there another workaround?

**Comment by drolin-suffolk** [comment]
We're also seeing another issue with the proposed work around.  We're using the `applyProjectChanges(...)` method to keep clients in sync.  That method throws an exception on project change:
```
Cannot destructure property 'store' of 'me.getStoreDescriptor(...)' as it is null
```
This may be related to [5328](https://github.com/bryntum/support/issues/5238), as we're using the working around on that ticket, and there is a comment right above the line that throwing the exception, referencing 5328.

**Comment by arcady-zherdev** [comment]
We'll try to address this in the next patch release.

**Comment by arcady-zherdev** [comment]
> @arcady-zherdev Thanks for the snippet but upon implementing this, we found that it does not match the expected sync response in the format:
> 
> ```
> project : { updated/added/removed: [{startDate: Date}]}
> ```
> 
> Instead, we get back the entire project object in the sync response. Is there another workaround?

Multiple projects are not supported at the moment. A store reports if a model is added or removed which is impossible in this case since there is no `Store` keeping `ProjectModel` instances. So `added/updated/removed` structure makes no sense.

But your application of course can send data in any form you wish (use `beforeSync` listener or override the project `encode` method  to change requests format).

**Comment by arcady-zherdev** [comment]
> We're also seeing another issue with the proposed work around. We're using the `applyProjectChanges(...)` method to keep clients in sync. That method throws an exception on project change:
> 
> ```
> Cannot destructure property 'store' of 'me.getStoreDescriptor(...)' as it is null
> ```
> 
> This may be related to [5328](https://github.com/bryntum/support/issues/5238), as we're using the working around on that ticket, and there is a comment right above the line that throwing the exception, referencing 5328.

Ok I'll check what happens
